vividus-plugin-web-app/src/main/java/org/vividus/steps/ui/web/WindowSteps.java

```diff
@@ -49,10 +66,96 @@ public void resizeCurrentWindow(Dimension targetSize)
     {
         if (canTryToResize(targetSize))
         {
-            webDriverProvider.get().manage().window().setSize(targetSize);
+            getWebDriver().manage().window().setSize(targetSize);
         }
     }
 
+    /**
+     * Closes <b>current window</b> and switches to the window from which rederection to current window was performed
+     * <p>
+     * Each browser <b>window</b> or <b>tab</b> is considered to be a separate <b>window object</b>. This object holds
+     * corresponding <b>Document</b> object, which itself is a html page. So this method applies to both windows and
+     * tabs.
+     * <p>
+     * Actions performed at this step:
+     * <ul>
+     * <li>Receives all opened browser windows
+     * <li>Identifies current window and closes it
+     * <li>Switches back to the window from which rederection to current window was performed
+     * </ul>
+     * @see <a href="https://html.spec.whatwg.org/#browsing-context"><i>Browsing context (Window &amp; Document)</i></a>
+     * @see <a href="https://www.w3schools.com/tags/default.asp"><i>HTML Element Reference</i></a>
+     */
+    @When("I close the current window")
+    public void closeCurrentWindow()
+    {
+        tryToCloseCurrentWindow((driver, window) ->
+        {
+            driver.close();
+            driver.switchTo().window(window);
+        });
+    }
+
+    /**
+     * Trying to close the <b>current window</b> with JavaScript method 'close()'.
+     * If an alert window via 'onbeforeunload' event is opened, it must be checked and handled in the subsequent steps.
```

Comment:
If an alert window is opened via 'onbeforeunload' ...
imho it's more clear to read